### PR TITLE
Update azure-autorust to append path and parameters via URL lib

### PR DESCRIPTION
**Pull request by andras-pinter**
When returning a URL from Client::endpoint the URL contains a tailing slash in any case. This is handled by the URL lib which will append the tailing slash to the end upon calling Display::fmt. Therefore formatting the returned URL from Client::endpoint will result in a duplicated slash, which could cause some (not all) Azure endpoints to have some trouble.

For example:
Base URL: https://some-app-conf.azconfig.io/
AppConfigUrl: https://some-app-conf.azconfig.io//keys?api-version=2023-10-01 This resulted in a 404 response.

Removing the tailing slash from the base endpoint's URL manually does not solve the problem, due to the reason mentioned above.

This patch applies a fix to append (set) the formatted path via rust URL lib, which will handle the duplicated slash (or any other problem).

Although I tried to locally generate `mgmt` and `svc` crates, I encountered some problems, and `autorust` removed and added a bunch of new files and modules.
So I'd like to ask somebody to help me out with the generation.

### CODE REVIEW

**Comment by andras-pinter** [comment]
Hey @heaths,

Yes, there is a particular error, what I recently encountered. Please see:
https://github.com/Azure/azure-sdk-for-rust/issues/1664

**Comment by heaths** [comment]
@ctaggart could you review this? If you're good with it, I can regenerate the service crates.

**Comment by andras-pinter** [comment]
My goal was to let `azure_core::Url` handle URL creation :)

File: services/autorust/codegen/src/codegen_operations/request_builder_send.rs [hunk 1]
```diff
@@ -58,7 +57,8 @@ impl ToTokens for RequestBuilderSendCode {
             let api_version = &self.request_builder.api_version;
             quote! {
                 fn url(&self) -> azure_core::Result<azure_core::Url> {
-                    let mut url = azure_core::Url::parse(&format!(#fpath, self.client.endpoint(), #url_str_args))?;
+                    let mut url = self.client.endpoint().clone();
+                    url.set_path(&format!(#fpath, #url_str_args));
```
Comment (cataggar) [inline services/autorust/codegen/src/codegen_operations/request_builder_send.rs:61]: This change looks good to me.

**Review by cataggar — APPROVED** [review]

**Comment by cataggar** [comment]
> @ctaggart could you review this? If you're good with it, I can regenerate the service crates.

Sounds good. Please go ahead.

**Comment by kingofthehill444** [comment]
I'm facing this exact issue when trying to use the azure web pub sub crate. 

I'm happy to help out but it seems like the change just needs to be merged? Can someone help push this through please?

**Comment by andras-pinter** [comment]
I made this as a PR instead of a draft. Still facing some issues, if somebody please help me finish this would be really nice! Thanks!

**Comment by heaths** [comment]
Most of the issues were with nightly tests failing due to evolving lints. Checking if a rebase on `main` fixes them first.

**Review by heaths — APPROVED** [review]

**Comment by andras-pinter** [comment]
@microsoft-github-policy-service agree

**Comment by andras-pinter** [comment]
🎉  Thanks everyone!

**Pull request merged**